### ISSUE TITLE
Think about the front page

### ISSUE THREAD

**Issue by riceissa** [vipulnaik/donations#35]
Currently the front page itself is ~24MB. This is big/slow enough that it's pretty annoying to visit it; in fact, I find it so annoying that I almost never visit it.

I would prefer a landing page that was quick to load and gave me e.g. top-30 stuff (top-30 is consistent with graphs) and links to things I would actually want to see. There could be a separate page with the full list for people who want to see that.

**Comment by riceissa** [comment]
I'm working on this at https://github.com/vipulnaik/donations/tree/faster-front-page

**Comment by riceissa** [comment]
```bash
$ curl 'http://localhost:8000/' | wc -c
324691
$ curl 'https://donations.vipulnaik.com/' | wc -c
32804584
```

So right now I've gotten it down to be about two orders of magnitude smaller. The loading time is still slow though, I think possibly because of the subqueries I have introduced (as part of the the limiting process). But that could be solved by caching the page? Alternatively, it might turn out to be quite fast on the server.

**Comment by riceissa** [comment]
It looks like in graphs, "Total" is counted as one of the 30, so actually we want the top-29 so that with "Total" there are 30.

**Comment by riceissa** [comment]
As for loading time, `time curl --silent http://localhost:8000 | wc -c` shows that the master branch takes about 37 seconds while faster-front-page takes about 31 seconds (both on my laptop). I ran the command three times on each branch and averaged the second and third runs. Of course, this ignored network transfer.

**Comment by riceissa** [comment]
The full front page can now be retrieved with `display=full`, i.e. <donations.vipulnaik.com/?display=full> (once the branch is merged).

**Comment by riceissa** [comment]
I think I will add `display=individual` as an option to deal with #36.

**Comment by riceissa** [comment]
I actually decided to call it `display=individual-scale` to include both "individual" and "couple" (and possibly other, in the future) `donor_type`s. (I think `display=individual` would have been confusing, because the database meaning and URL parameter meaning would be different, although I guess we've already done that with cause area.)

**Comment by riceissa** [comment]
```bash
$ for i in 1 2 3; do (time curl --silent http://localhost:8000 | wc -c); done
324601

real    0m31.291s
user    0m0.021s
sys     0m0.014s
324601

real    0m30.712s
user    0m0.015s
sys     0m0.012s
324601

real    0m30.987s
user    0m0.018s
sys     0m0.011s
$ for i in 1 2 3; do (time curl --silent http://localhost:8000/?display=full | wc -c); done
32805743

real    0m46.477s
user    0m0.049s
sys     0m0.123s
32805743

real    0m38.187s
user    0m0.039s
sys     0m0.114s
32805743

real    0m40.176s
user    0m0.064s
sys     0m0.087s
$ for i in 1 2 3; do (time curl --silent http://localhost:8000/?display=individual-scale | wc -c); done
246622

real    0m39.657s
user    0m0.022s
sys     0m0.018s
246622

real    0m19.655s
user    0m0.019s
sys     0m0.016s
246622

real    0m18.247s
user    0m0.021s
sys     0m0.011s
```

**Comment by riceissa** [comment]
There was a bug in the individual-scale one, where I was appending the restriction to the wrong variable. It runs much faster now:

```bash
$ for i in 1 2 3; do (time curl --silent http://localhost:8000/?display=individual-scale | wc -c); done
230981               

real    0m4.090s     
user    0m0.028s     
sys     0m0.005s     
230981               

real    0m4.066s     
user    0m0.024s     
sys     0m0.007s     
230981               

real    0m4.090s     
user    0m0.023s     
sys     0m0.008s
```

**Comment by riceissa** [comment]
I think this is good enough to be merged to master now, so I'm passing it along to @vipulnaik for review.

**Comment by riceissa** [comment]
I should note that part of the reason `display=individual-scale` is so small right now is because many individuals are not yet entered in the `donors` table (i.e. they only exist as donors in the `donations` table right now). We need the donor in the `donors` table because that is how we tell a donor is an individual. `select count(distinct donor) from donations` gives 334 donors while `select count(distinct donor) from donors where donor_type = 'Individual' or donor_type = 'Couple'` gives 52.

**Comment by riceissa** [comment]
I added output buffering caching in the cache-pages branch.

**Comment by riceissa** [comment]
I'm closing this now, since the front page is smaller and much faster now.
